sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -190,6 +190,11 @@ trait StateStore extends ReadStateStore {
   /** Current metrics of the state store */
   def metrics: StateStoreMetrics
 
+  /** Return information on recently generated checkpoints */
+  def getCheckpointInfo: StateStoreCheckpointInfo = {
+    StateStoreCheckpointInfo(-1, -1, None, None)
```

Comment:
Why default implementation? If all the sub-classes are overriding it, let's just make it required with no default.

Reply:
+1, it could lead to bugs if this is incorrect, right? I'd remove a default implementation in such a case (it may require changes in tests I guess, but that can be handled with a trait or something)